app/models/pipeline_run.rb

```diff
@@ -411,6 +424,60 @@ def db_load_ercc_counts
     update(total_ercc_reads: total_ercc_reads)
   end
 
+  def db_load_insert_size_metrics
+    insert_size_metrics_s3_path = "#{host_filter_output_s3_path}/#{INSERT_SIZE_METRICS_OUTPUT_NAME}"
+    _stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", insert_size_metrics_s3_path)
+    return unless status.exitstatus.zero?
+    insert_size_metrics_lines = Syscall.pipe_with_output(["aws", "s3", "cp", insert_size_metrics_s3_path, "-"])
+    tsv_lines = []
+    tsv_header_line = -1
+    insert_size_metrics_lines.split(/\r?\n/).each_with_index do |line, index|
+      if line.start_with?("## METRICS CLASS")
+        tsv_header_line = index
+      elsif tsv_header_line > 0 && index - tsv_header_line <= 2
+        tsv_lines << CSV.parse_line(line, col_sep: "\t")
+      elsif tsv_lines.length >= 2
+        break
+      end
+    end
+    if tsv_lines.length != 2
+      Rails.logger.error("Pipleine run ##{id} has an insert size metrics file but metrics could not be found")
+      return
+    end
+    insert_size_metrics = {}
+    tsv_lines[0].zip(tsv_lines[1]).each do |row|
+      insert_size_metrics[row[0]] = row[1]
+    end
+
+    extract_int = lambda do |metrics, metric_name|
+      begin
+        return metrics[metric_name].to_i
+      rescue
+        return nil
+      end
+    end
+
+    insert_size_median = extract_int.call(insert_size_metrics, MEDIAN_INSERT_SIZE_NAME)
```

Comment:
Do you need nil here if the key is not present or is zero is acceptable in this case?

`insert_size_median = insert_size_metrics[MEDIAN_INSERT_SIZE_NAME].to_i` would assign zero if the key is missing.

Let me know if you need `nil` instead and I can show you some patterns

Reply:
I do need `nil` sorry, I explained my my earlier comment before I saw this. I have this so far:

```ruby
  private def extract_int_metric(metrics, metric_name)
    if metrics[metric_name]
      return metrics[metric_name].to_i
    end
    return nil
  end
```

Is that better?